src/datatypes/src/vectors/binary.rs

```diff
@@ -36,6 +36,40 @@ impl BinaryVector {
     pub(crate) fn as_arrow(&self) -> &dyn Array {
         &self.array
     }
+
+    /// Creates a new binary vector of JSONB from a binary vector.
+    /// The binary vector must contain valid JSON strings.
+    pub fn convert_binary_to_json(&self) -> Result<BinaryVector> {
+        let arrow_array = self.to_arrow_array();
+        let mut vector = vec![];
+        for binary in arrow_array
+            .as_any()
+            .downcast_ref::<BinaryArray>()
+            .unwrap()
+            .iter()
+        {
+            let jsonb = if let Some(binary) = binary {
+                if is_jsonb(binary) {
+                    Some(binary.to_vec())
```

Comment:
Here's an extra copy if the conversion has already been made, e.g. INSERT INTO table VALUES(parse_json("1")). Since `BinaryArray` can directly take `&[u8]` to construct, I wonder if we can avoid the copy here.